### PR TITLE
feat(cb2-7050): improve lambda efficiency

### PR DESCRIPTION
**Pull request by aaron-downing**
## Data transfer efficiency: Refactor of EDH-Marshaller Lambda 

There were a number of issues in the marshaller lambda that made processing DynamoDB stream events sluggish. The following has been changed:
- Target SQS URLs are now retrieved from environment variables set via terraform instead of making async calls to retrieve the URL from AWS.
- SQS client is cached between invocations
- Removed .yaml file that was used for deciding SQS target as it was redundant and causing performance issues
- Intermittent failures are now retried using BatchItemFailures as a response

The above change now means that an invocation takes on average 700ms instead of previously 2600ms.

Before change:
image

After change:
image

[CB2-7050](https://dvsa.atlassian.net/browse/CB2-7050)

## Checklist

- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/functions/edh-marshaller.ts [hunk 1]
```diff
@@ -3,72 +3,119 @@ import {
...
+  const sendMessagePromises = [];
 
+  if (!sqs) {
+    sqs = AWSXray.captureAWSClient(
+      new SQS({
+        region: 'eu-west-1',
+        apiVersion: '2012-11-05',
+      }),
+    );
+  }
```
Comment (tahayazici-bjss) [inline src/functions/edh-marshaller.ts:49]: Maybe this is a bit simpler? Saves you the `if` nesting.
```suggestion
  sqs = sql ?? AWSXray.captureAWSClient(
    new SQS({
      region: 'eu-west-1',
      apiVersion: '2012-11-05',
    }),
  );
```

File: src/functions/edh-marshaller.ts [hunk 2]
```diff
@@ -3,72 +3,119 @@ import {
...
+    if (!params) {
+      console.error(`Unable to generate SQS event for record: ${JSON.stringify(record)}`);
+      continue;
+    }
+
+    sendMessagePromises.push(
+      sqs.sendMessage(params).promise()
+        .then(() => {
+          debugOnlyLog('Succesfully sent SQS message');
+        })
```
Comment (tahayazici-bjss) [inline src/functions/edh-marshaller.ts]: Single line:
```suggestion
        .then(() => debugOnlyLog('Succesfully sent SQS message'))
```

File: tests/unit/functions/edh-marshaller.test.ts [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
+    mSendMessage(params);
+
+    return { 
+      promise: jest.fn().mockImplementation(() => {
+        if (params.QueueUrl === 'FAIL') {
+          return Promise.reject(<PromiseResult<SQS.SendMessageResult, AWSError>>{});
+        }
+        return Promise.resolve(<PromiseResult<SQS.SendMessageResult, AWSError>>{});
+      }),
+    };
```
Comment (tahayazici-bjss) [inline tests/unit/functions/edh-marshaller.test.ts]: Just to make it a bit cleaner:
```suggestion
      promise: jest.fn().mockImplementation(() => params.QueueUrl === 'FAIL'
        ? Promise.reject(<PromiseResult<SQS.SendMessageResult, AWSError>>{})
        : Promise.resolve(<PromiseResult<SQS.SendMessageResult, AWSError>>{})),
    };
```

**Review by tahayazici-bjss — COMMENTED** [review]
Great stuff, left a few comments to simplify.

File: src/functions/edh-marshaller.ts (hunk 1 above)
Comment (aaron-downing) [reply src/functions/edh-marshaller.ts:49]: In terms of readability I think I prefer the if check as with the null coalesce it feels weird doing self assignment if 
`sqs` already defined.

**Review by tahayazici-bjss — DISMISSED** [review]

File: src/utils/utilities.ts [hunk 4]
```diff
@@ -2,27 +2,12 @@
...
-import { Configuration } from './config';
-import { ERROR } from '../models/enums';
 
 export const getTechRecordEnvVar = (): boolean => {
-  if (process.env.PROCESS_FLAT_TECH_RECORDS == 'false') {
-    return false;
-  } else if (process.env.PROCESS_FLAT_TECH_RECORDS == 'true') {
+  if (process.env.PROCESS_FLAT_TECH_RECORDS == 'true') {
     return true;
   }
```
Comment (gjulien-bjss) [inline src/utils/utilities.ts]:
```suggestion
 return process.env.PROCESS_FLAT_TECH_RECORDS === 'true'
```
(without the `return false` on line 10 as well)

Comment (aaron-downing) [reply src/utils/utilities.ts]: Nice 😎

**Review by mcadenas-bjss — APPROVED** [review]

**Review by Ian-Hodges — APPROVED** [review]

**Pull request merged**